alchemy/src/prisma/database.ts

```diff
@@ -0,0 +1,251 @@
+   */
+  region?:
+    | "us-east-1"
+    | "us-west-1"
+    | "eu-west-3"
+    | "ap-northeast-1"
+    | "ap-southeast-1";
+
+  /**
+   * Whether this is the default database for the project
+   * @default false
+   */
+  isDefault?: boolean;
+}
+
+/**
+ * A Prisma database connection/API key
+ */
+export interface DatabaseConnection {
+  /**
+   * Connection ID
+   */
+  id: string;
+
+  /**
+   * Connection name
+   */
+  name: string;
+
+  /**
+   * Time at which the connection was created
+   */
+  createdAt: string;
+
+  /**
+   * Database connection string
+   */
+  connectionString: string;
+}
+
+/**
+ * Output returned after Prisma database creation/update
+ */
+export interface Database extends Resource<"prisma::Database"> {
+  /**
+   * The ID of the database
+   */
+  id: string;
+
+  /**
+   * The ID of the project this database belongs to
+   */
+  projectId: string;
+
+  /**
+   * Name of the database
+   */
+  name: string;
+
+  /**
+   * Region where the database is deployed
+   */
+  region: string | null;
+
+  /**
+   * Whether this is the default database for the project
+   */
+  isDefault: boolean;
+
+  /**
+   * Database connection string (only available during creation)
+   */
+  connectionString?: string;
+
+  /**
+   * Database status
+   */
+  status?: string;
+
+  /**
+   * API keys/connections for this database
+   */
+  apiKeys?: PrismaDatabaseConnection[];
+
+  /**
+   * Time at which the database was created
+   */
+  createdAt: string;
+}
+
+/**
+ * Creates a Prisma database within a project for data storage and management.
+ *
+ * @example
+ * ## Create a basic database
+ *
+ * ```ts
+ * const database = await Database("my-database", {
+ *   project: project,
+ *   name: "my-app-db",
+ *   region: "us-east-1"
+ * });
+ * ```
+ *
+ * @example
+ * ## Create a default database
+ *
+ * ```ts
+ * const database = await Database("default-db", {
+ *   project: "project-123",
+ *   name: "production",
+ *   region: "us-east-1",
+ *   isDefault: true
+ * });
+ * ```
+ *
+ * @example
+ * ## Create database with custom region
+ *
+ * ```ts
+ * const database = await Database("eu-database", {
+ *   project: project,
+ *   name: "eu-production",
+ *   region: "eu-west-3"
+ * });
+ * ```
+ */
+export const Database = Resource(
+  "prisma::Database",
+  async function (
+    this: Context<Database>,
+    id: string,
+    props: DatabaseProps,
+  ): Promise<Database> {
+    const api = createPrismaApi(props);
+    const projectId =
+      typeof props.project === "string" ? props.project : props.project.id;
+    const databaseId = this.output?.id;
+
+    if (this.phase === "delete") {
+      try {
+        if (databaseId) {
+          const deleteResponse = await api.delete(
+            `/projects/${projectId}/databases/${databaseId}`,
+          );
+          if (!deleteResponse.ok && deleteResponse.status !== 404) {
+            await handleApiError(deleteResponse, "delete", "database", id);
+          }
```

Comment:
pull these kinds of functions out into helpers, `async function deleteDatabase(api, ..)`